lua/roslyn/init.lua

```diff
@@ -203,7 +207,7 @@ function M.setup(config)
 
             local exe = roslyn_config.exe
             local mason_installation = get_mason_installation()
-            if not vim.uv.fs_stat(exe) and not vim.uv.fs_stat(mason_installation) then
+            if not vim.uv.fs_stat(exe) and not vim.uv.fs_stat(mason_installation) and not vim.fn.executable(exe) then
```

Comment:
I don't think this is necessary. If this is evaluated, then the file doesn't exist. Then it can't be an executable. Also, this returns 0 or 1 I believe, so this will always evaluate to true(?). So I think we can just remove this

Reply:
ah, quite possibly. Without this though, when I pass in "Microsoft.CodeAnalysis.LanguageServer" as `exe`, the `vim.uv.fs_stat(exe)` check fails, since "Microsoft.CodeAnalysis.LanguageServer" is not a file.

With this new check, it successfully finds and asserts "Microsoft.CodeAnalysis.LanguageServer" as an executable and happily continues.

I'll be completely honest though, I'm not entirely sure how `vim.fn.executable` works, just found it after some cursory googling :)

I'll push up my neovim config in a moment as a working example.

Reply:
btw, amazing effort on getting this to work! Made my life so much easier